internal/controller/pod_controller.go

```diff
@@ -168,3 +189,84 @@ func (a *PodAdmissionWebhook) listOwners(ctx context.Context, object client.Obje
 	}
 	return owners, nil
 }
+
+type podDeleteController struct {
+	client.Client
+	Scheme  *runtime.Scheme
+	updater *workload.Updater
+}
+
+// newDeletePodController constructs an podDeleteController
+func newPodDeleteController(mgr ctrl.Manager, u *workload.Updater) (*podDeleteController, error) {
+	r := &podDeleteController{
+		Client:  mgr.GetClient(),
+		Scheme:  mgr.GetScheme(),
+		updater: u,
+	}
+	err := r.setupWithManager(mgr)
+	return r, err
+}
+
+// setupWithManager adds this AuthProxyWorkload controller to the controller-runtime
+// manager.
+func (r *podDeleteController) setupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Pod{}).
+		Complete(r)
+}
+
+func (r *podDeleteController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Read the ReplicaSet
+	pod := &corev1.Pod{}
+	err := r.Client.Get(ctx, req.NamespacedName, pod)
+	if err != nil {
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+	// Pod is already deleted, ignore this change event.
+	if !pod.ObjectMeta.DeletionTimestamp.IsZero() {
+		return reconcile.Result{}, nil
+	}
+
+	err = r.handlePodChanged(ctx, pod)
+	if err != nil {
+		return reconcile.Result{Requeue: true}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// handlePodChanged Deletes pods that meet the following criteria:
+// 1. The pod is in Error or CrashLoopBackoff state.
+// 2. The pod matches one or more AuthProxyWorkload resources.
+// 3. The pod is missing one or more proxy sidecar containers for the resources.
+func (r *podDeleteController) handlePodChanged(ctx context.Context, pod *corev1.Pod) error {
+
+	wl := &workload.PodWorkload{Pod: pod}
+
+	// Find all proxies that match this pod
+	proxies, err := findMatchingProxies(ctx, r.Client, r.updater, wl)
+	if err != nil {
+		return fmt.Errorf("unable to find proxies when pod changed, %v", err)
+	}
+
+	// There are no proxies, ignore this event.
+	if len(proxies) == 0 {
+		return nil
+	}
+
+	// Check if this pod is in error and missing proxy containers
```

Comment:
add detail here? is `in error` the proper wording? `in error state` perhaps? 
```suggestion
	// Check if this pod is in error state and missing proxy containers
```

Reply:
fixed.